master/buildbot/reporters/zulip.py

```diff
@@ -0,0 +1,58 @@
+from twisted.internet import defer
+
+from buildbot import config
+from buildbot.reporters import utils
+from buildbot.reporters.http import HttpStatusPushBase
+from buildbot.util import httpclientservice
+from buildbot.util.logger import Logger
+
+logger = Logger()
+
+
+class ZulipStatusPush(HttpStatusPushBase):
+    name = "ZulipStatusPush"
```

Comment:
Was there a particular reason why more of the `HTTPStatusPush` functionality has not been used? Can Zulip support additional properties like those supported in `utils.getDetailsForBuild`? If that's the case, I think it would be great if we allowed to send that data too.